router/core/operation_planner.go

```diff
@@ -53,6 +53,20 @@ func (p *OperationPlanner) preparePlan(ctx *operationContext) (*planWithMetaData
 		return nil, &reportError{report: &report}
 	}
 
+	// The doc is parsed using ctx.content which is the normalized content from the original parsed content
+	// This contains the original operation name, so we need to manually tell the parser to ignore the
+	// extra parts
+	for i := range doc.RootNodes {
+		if doc.RootNodes[i].Kind != ast.NodeKindOperationDefinition {
+			continue
+		}
+		currName := doc.OperationDefinitionNameString(doc.RootNodes[i].Ref)
+		if trimmed, _ := GetOperationNameTrimmed(currName, ctx.operationTrimLimit); trimmed {
+			name := doc.OperationDefinitions[doc.RootNodes[i].Ref].Name
+			doc.OperationDefinitions[doc.RootNodes[i].Ref].Name.End = name.Start + uint32(ctx.operationTrimLimit)
+		}
+	}
```

Comment:
Does this only apply to Root nodes? Could this happen with a long alias, directive, variable maybe? Isn't anything ugc that ends up in metrics vulnerable?